plugins/flutter_aepedge/test/flutter_aepedge_test.dart

```diff
@@ -44,4 +46,148 @@ void main() {
+          arguments: experienceeventData,
+        ),
+      ]);
+    });
+
+    test('returns correct result', () async {
+      final actualEventHandleResponse = await Edge.sendEvent(experienceEvent);
+      expect(actualEventHandleResponse[0].payload, expectedResponse[0].payload);
+      expect(actualEventHandleResponse[0].type, expectedResponse[0].type);
+    });
+  });
+
+  group('sendEvent nested', () {
+    final Map<dynamic, dynamic> mapValue = {"keySample": "keyValue"};
+    final Map<dynamic, dynamic> dataValue = {"keySample1": "keyValue1"};
+    final Map<dynamic, dynamic> xdmData = {"eventType": mapValue};
+    final Map<String, dynamic> data = {"free": "form", "data": dataValue};
+
+    final Map<String, dynamic> experienceeventData = {
+      "xdmData": xdmData,
+      "data": data,
+      "datasetIdentifier": "datasetExample",
+    };
+
+    //setup experienceEvent
+    final ExperienceEvent experienceEvent =
+        ExperienceEvent(experienceeventData);
+
+    final String eventHandleType = "state:store";
+    final List<dynamic> eventHandlePayload = [
+      {"maxAge": 1800, "value": "testValue1", "key": "keyExample1"},
+      {"maxAge": 34128000, "value": "testValue2", "key": "keyExample2"}
+    ];
+
+    final Map<dynamic, dynamic> eventHandleExpectedData = {
+      "type": eventHandleType,
+      "payload": eventHandlePayload
+    };
+
+    final String eventHandleType2 = "state:store";
+    final List<dynamic> eventHandlePayload2 = [
+      {
+        "type": "samplestore#",
+        "destinationId": "345",
+        "alias": "auto",
+        "segments": [
+          {"id": "00001234-7nfj-78u0-0ne7-nju348098jd1"},
+          {"id": "78947208-yuj6-78nh-672m-k8792d7v9wnb"}
+        ]
+      },
+    ];
+
+    final Map<dynamic, dynamic> eventHandleExpectedData2 = {
+      "type": eventHandleType2,
+      "payload": eventHandlePayload2
+    };
+
+    //setup returned and expected EventHandles
+    final EventHandle expectedEvent = EventHandle(eventHandleExpectedData);
+    final EventHandle returnedEvent = EventHandle(eventHandleExpectedData);
+
+    final EventHandle expectedEvent2 = EventHandle(eventHandleExpectedData2);
+    final EventHandle returnedEvent2 = EventHandle(eventHandleExpectedData2);
+
+    final List<EventHandle> expectedResponse = [expectedEvent, expectedEvent2];
+    final List<dynamic> returnedResponse = [
+      returnedEvent.data,
+      returnedEvent2.data
+    ];
+
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return returnedResponse;
+      });
+    });
+    test('invokes correct method happy test', () async {
+      await Edge.sendEvent(experienceEvent);
+
+      expect(log, <Matcher>[
+        isMethodCall(
+          'sendEvent',
+          arguments: experienceeventData,
+        ),
+      ]);
+    });
+
+    test('returns correct result', () async {
```

Comment:
We should assert implementationDetails from the XDM data from the experience event sent here.

Reply:
As discussed, will validate this in the manual testing.